daemon/files.py

```diff
@@ -188,38 +178,33 @@ def ports(self, ports: str):
         except ValueError:
             self._logger.warning(f'invalid value `{ports}` passed for \'ports\'')
 
-    @cached_property
+    @property
     def requirements(self) -> str:
         """pip packages mentioned in requirements.txt
 
         :return: space separated values
         """
-        # TODO: merge this with _read_requirements_file()
-        _req = f'{self._workdir}/requirements.txt'
-        if not Path(_req).is_file():
+        requirements = ''
+        for filename in os.listdir(self._workdir):
+            if is_requirements_txt(filename):
+                with open(os.path.join(self._workdir, filename)) as f:
+                    requirements += ' '.join(f.read().splitlines())
```

Comment:
do we have something similiar (req file parsing) maybe for hubble? I was just wondering, because this will miss special cases like `-f https://some-custom-repo.com`

Reply:
Hmm. We only have tested with requirements that have 1 word in them (e.g. - `jina`, `git+...`). If need arises, let us change the logic in the dockerfile. Rather than passing the requirements as an argument during `docker build`, let's pass names of the requirements files as an argument and let `pip install -r` handle the test.